### DISCUSSION TITLE
Looking for examples of working with tp.obsidian object

### DISCUSSION THREAD

**Discussion by snovis** [SilentVoid13/Templater#1087] [Help] [answered]
From the latest documentation, it seems that template gives access to the Obsidian API via `tp.obsidian` object, but I can not find a single example of working code that does this.

Specifically I'm trying to access editor functions.  I found work around here: https://github.com/SilentVoid13/Templater/discussions/419 where the author used the App object to access the editor.  However, I'm curious to see how the tp.obsidian object is supposed to work.

**Comment by Zachatoo** [comment]
I have two examples of using the `tp.obsidian`, [here](https://zachyoung.dev/posts/templater-snippets#suggester-for-files-with-tag) and [here](https://zachyoung.dev/posts/templater-snippets#get-most-recently-modified-file-with-specific-tag).

You really only should use `tp.obsidian` for anything in the Obsidian API that is exported globally without a namespace. You can tell what those are by searching for `export function` in the [Obsidian API definitions](https://github.com/obsidianmd/obsidian-api/blob/master/obsidian.d.ts). Editor is not one of those things.

To access the editor, you'll need to do what that other post is doing, accessing it off of `app.workspace.activeLeaf.view.editor`. You do not need `tp.obsidian` to do that.

**Reply by snovis** [reply]
thank you taking the time to answer my question - that feels so counter intuitive!  I would have expected the obsidian object to contain obsidian features like the editor.  Do you know where I can find a document that explains or gives an overview of that relationship?  In other words, HOW did you learn that?  (I suppose my blindspot is not knowing/appreciating the difference between the app and obsidian)
thanks!

**Reply by Zachatoo** [reply, marked as answer]
I found out how to use `tp.obsidian` by typing `<%* console.log(tp.obsidian) %>` into a note and running that template to see what I had access to on the `tp.obsidian` object. Did some trial and error by calling properties on that object, found that most things that start with an uppercase letter (like `App`) were unusable, probably because they're classes and I don't know what Obsidian passes to them. Everything that started with a lowercase letter (like `getAllTags`) worked just fine.

I knew about how to use `app` because I've developed a few Obsidian plugins and knew some of the properties available on `app` already, though you can do a similar thing with `<%* console.log(app) %>`, or even easier, you can open the dev tools console (`cmd shift i` or `ctrl shift i`) and type `app` in the console.

**Reply by snovis** [reply]
Awesome reply.  Thank you VERY much!
